test/test_jit.py

```diff
@@ -3970,6 +3970,49 @@ def foo():
 
         self.assertEqual(foo(), [1, 2, 3, 4])
 
+    def test_extend_list_mutable(self):
+        @torch.jit.script
+        def extend_list(a, b):
+            # type: (List[Tensor], List[Tensor]) -> List[Tensor]
+
+            a.extend(b)
+            return a
+
+        for l in [[], [torch.rand(2)], [torch.rand(2), torch.rand(2), torch.rand(2)]]:
+            for r in [[], [torch.rand(2)], [torch.rand(2), torch.rand(2), torch.rand(2)]]:
+                rl = l[:]  # side effects due to extend
+                self.assertEqual(extend_list(l, r), rl + r)
```

Comment:
Can we also verify that `rl` has been mutated?

Reply:
I don't believe, `rl` would change, but `rl + r` would give us the correct answer to compare against `l`. Why do we want it to be mutated?

Reply:
actually, I can switch the order of args in `assertEqual` that we shouldn't need a copy.

Reply:
@apaszke scratch that. JIT makes a copy of every `list` argument meaning we never modify `l`.